### ISSUE TITLE
Research trajectory-derived Skill acquisition with held-out evaluation

### ISSUE THREAD

**Issue by ist-h-i** [ist-h-i/agent-spectrum-kernel#305]
Parent roadmap: #170
Research dependency for: #176, #278
Evaluation authority: #197
Asset / Portfolio foundations: #276, #277
Potential later portability consumer: #180

## Problem

ASK can already register, select, evaluate, evolve, roll back, and retire exact AI Engineering Assets. It also has a project-local finding-to-candidate direction in #176.

What is not established is whether ASK can **learn a genuinely reusable new Skill from its own task execution experience** rather than merely restating what the model already knew.

There is a bootstrap risk:

- a model strong enough to infer the correct Skill from a task may already solve that task without the Skill;
- a weaker model may generate an incorrect heuristic and persist it as durable guidance;
- evaluating on the same tasks used to generate the Skill can mistake memorization or overfitting for learning;
- a generated Skill can improve one metric while increasing false confidence, token cost, latency, maintenance cost, or regressions elsewhere.

The research question is therefore not whether an LLM can write a `SKILL.md`. It is whether verified execution experience can be converted into a reusable capability that improves future unseen tasks enough to justify becoming an ASK Asset.

## Goal

Run a bounded, result-blind experiment that determines whether verified task trajectories can produce **trajectory-derived Skill candidates** that improve held-out tasks from the same task distribution.

The first experiment must treat generated Skills as hypotheses only. It must not make self-generated Skill acquisition a standard ASK capability unless the measured result supports that decision.

### H1 — trajectory-derived capability

A Skill candidate generated from verified successful/failed task trajectories can improve a held-out task set relative to the same model/harness using the current Portfolio with no newly generated Skill.

### H2 — selection matters

Generating multiple materially distinct candidates and selecting them through held-out evaluation produces better generalization than generating one candidate and accepting it directly.

### H3 — value must survive total cost

Any observed quality benefit remains useful after accounting for:

- token and model cost;
- latency;
- tool calls / execution work;
- human correction or review;
- false-positive / unnecessary intervention;
- Skill generation and evaluation cost;
- maintenance and refresh cost where measurable.

A neutral, harmful, overfit, or insufficient result is a valid outcome.

## First bounded vertical

Use **Skill Assets only** for the first experiment.

Do not broaden the first protocol to Prompt, Rule, Router, Portfolio policy, setup default, executable check, or other Asset classes. Those may become follow-up hypotheses only if the Skill vertical demonstrates a useful acquisition signal.

Choose one stable task family where ASK has repeatable evidence and representative variation. A PR-review / engineering-review family is a candidate, but the exact family must be frozen before result access.

The initial experiment should prefer the same or closely related repositories, stack, tools, and operator environment. This intentionally tests the high-reuse setting where the task distribution is relatively stable. Cross-repository and cross-organization transfer are separate questions.

## Inputs allowed for candidate derivation

The preregistered derivation set may include bounded, privacy-safe observations such as:

- task class and goal;
- selected Portfolio / Skill identities;
- summarized tool/action trajectory;
- success/failure and retry structure;
- verification results;
- review findings and corrections;
- human correction category without personnel-performance narrative;
- measured cost / token / latency data where available;
- exact source/evidence references.

Do not require durable storage of raw prompts, full conversations, secrets, customer data, personal data, private evaluator answers, or full source snapshots merely to generate a candidate.

## Experimental protocol

Before measured-result access:

1. **Freeze the task family and environment**
   - model and version;
   - agent/runtime/harness;
   - adapter/tool capability set;
   - relevant repository/task-class boundaries;
   - current baseline Portfolio;
   - evaluator authority and stop rules.

2. **Freeze a derivation / held-out split**
   - derivation trajectories are available to candidate generation;
   - held-out task outcomes and evaluator answers are not available to candidate generation;
   - no held-out task may be copied into Skill content.

3. **Define baselines**
   - current Portfolio / no newly generated Skill;
   - any other control required to distinguish extra reasoning from reusable Skill value.

4. **Generate multiple candidates**
   - produce at least 3 materially distinct Skill candidates when feasible;
   - candidates should encode different reusable hypotheses, not superficial rewrites;
   - preserve derivation provenance and exact content identity.

5. **Register as exploratory candidates only**
   - use #276 identity/provenance semantics;
   - registration is not validation, admission, activation, or demonstrated value.

6. **Freeze candidate comparison before results**
   - use #277/#278-compatible baseline/challenger identities;
   - candidate selection and evaluator scope must be sealed before measured outcomes are read.

7. **Evaluate on held-out tasks**
   - use reusable #197 evaluation authority where applicable;
   - run enough repetitions or sensitivity checks to distinguish one-off variance from a useful effect;
   - retain native-unit quality, cost, latency, human-effort, failure, and regression observations.

8. **Publish a bounded disposition**
   - `supported`;
   - `bounded`;
   - `neutral`;
   - `harmful`;
   - `insufficient_evidence`.

9. **Only after a supported/bounded result**
   - propose a separate production implementation Issue for automatic candidate acquisition;
   - define the exact applicability boundary learned from the experiment;
   - keep promotion / activation under existing #278 authority rather than creating a second lifecycle.

## Required comparisons

At minimum compare:

```text
Baseline:
  current Portfolio
  + same model/harness
  + no newly generated Skill

Challengers:
  same Portfolio/model/harness
  + candidate A
  + candidate B
  + candidate C ...
```

If extra test-time reasoning is used to generate or apply candidates, include a control sufficient to avoid attributing generic extra-compute gains to the persisted Skill itself.

## Evaluation dimensions

The preregistration must define exact metrics and decision thresholds before measured results are visible.

At minimum consider:

- task success / verification outcome;
- material review or implementation quality;
- missed-obligation / regression rate;
- false-positive or unnecessary-intervention rate;
- human correction / re-review effort;
- tokens and model cost;
- latency;
- tool calls / execution work;
- candidate-generation and evaluation cost.

Do not collapse all dimensions into one opaque score when native-unit tradeoffs matter.

## Generalization and overfitting controls

- Generation tasks and measured held-out tasks must be disjoint.
- Candidate generation must not access private evaluator answers for held-out tasks.
- A Skill that merely encodes one repository path, one patch, one issue answer, or one observed fixture is not reusable learning.
- Applicability and non-applicability conditions must be explicit.
- A candidate that improves the derivation set but not held-out tasks is rejected as non-generalizing.
- A candidate that improves average quality but creates a material safety or false-confidence regression is not automatically admitted.
- Results from one task family do not establish universal Skill-generation effectiveness.

## Research outputs

- result-blind preregistration;
- exact derivation / held-out inventory;
- baseline and candidate Asset identities;
- candidate-generation provenance;
- repeated held-out evaluation evidence;
- quality / cost / latency / human-effort comparison;
- generalization and regression analysis;
- final bounded disposition;
- recommendation to:
  - create a production acquisition capability;
  - run a narrower follow-up;
  - defer;
  - reject the hypothesis.

## Acceptance criteria

- [ ] The experiment is preregistered before measured held-out outcomes are accessed.
- [ ] Derivation and held-out sets are exact, disjoint, and reconstructable.
- [ ] At least one no-new-Skill/current-Portfolio baseline is measured with the same model/harness.
- [ ] Multiple materially distinct Skill candidates are generated when feasible; inability to do so is recorded rather than padded with rewrites.
- [ ] Candidate Assets preserve exact identity, derivation provenance, applicability, and exclusions.
- [ ] Generated candidates remain exploratory/challenger Assets and cannot self-promote.
- [ ] Held-out evaluation uses independent evaluator authority and prevents candidate access to hidden answers.
- [ ] Quality, regressions, false positives, total cost, latency, and human effort are visible in native units where available.
- [ ] The final result distinguishes supported, bounded, neutral, harmful, and insufficient evidence.
- [ ] A positive result creates evidence for a separate implementation decision; it does not silently enable production self-modification.
- [ ] A neutral/harmful result is allowed to terminate the feature direction without requiring a production implementation.
- [ ] GitHub Actions execution is not a completion criterion; validation evidence may come from another controlled environment.

## Stop conditions

Stop or narrow the experiment if:

- held-out contamination cannot be excluded;
- evaluator independence cannot be established;
- the runtime/model/environment cannot be frozen tightly enough for interpretable comparison;
- candidate generation requires private/customer data that cannot be represented safely;
- no representative repeated task family is available;
- the experiment cannot distinguish persisted-Skill value from generic extra test-time compute.

## Non-goals

- Production automatic Skill generation or activation
- Fully autonomous self-modification
- Automatic Portfolio promotion
- High-impact autonomous changes
- Model weight training or fine-tuning
- Hosted Skill marketplace or remote registry
- Cross-company/community/OSS distribution
- Proving all Asset classes can self-evolve
- Universal claims from one task family
- Replacing #176 project-learning lifecycle
- Replacing #197 evaluation, #276 Registry, #277 Portfolio, or #278 Evolution authority

## Relationship to existing ASK work

- **#176** remains the project-local finding-to-candidate lifecycle and may later consume a supported candidate-generation mechanism. It must not assume this research succeeds.
- **#197** remains the reusable evaluation/scoring authority; this Issue must not create a competing scoring stack.
- **#276** provides exact Skill Asset identity/provenance.
- **#277** provides result-blind baseline/challenger Portfolio identity and selection.
- **#278** remains the governed recommendation, approval, activation, rollback, and retirement authority.
- **#180** may later transport explicitly approved Assets, but portability does not imply effectiveness in a receiving environment. Imported Assets should remain local candidates until locally evaluated.
